geonode_mapstore_client/templates/geonode-mapstore-client/snippets/brand_navbar.html

```diff
@@ -106,45 +106,65 @@
                         menuBottomUl.appendChild(clone);
-                    } else {
-                        const items = clone.querySelectorAll(':scope > li');
-                        items.forEach(item => {
-                            const dropdownButton = item.querySelector('.dropdown button') || item.querySelector('.dropdown-toggle');
-                            if (dropdownButton) {
-                                // Create a new span with bold text
-                                const buttonText = dropdownButton.textContent.trim();
-                                const boldSpan = document.createElement('span');
-                                boldSpan.className = "dropdown-submenu-title";
-                                boldSpan.textContent = buttonText;
-                                
-                                // Add the bold span as a list item
-                                const headerLi = document.createElement('li');
-                                headerLi.appendChild(boldSpan);
-                                appendDivider();
-                                menuBottomUl.appendChild(headerLi);
-                                appendDivider();
-                                
-                                // Get all child items from the dropdown menu
-                                const childItems = item.querySelectorAll('.dropdown-menu > li') || item.querySelectorAll('ul > li');
-                                if (childItems) {
-                                    childItems.forEach(childItem => {
-                                        const _childItem = childItem.cloneNode(true);
-                                        _childItem.className = `${_childItem.className} dropdown-submenu`;
-                                        menuBottomUl.appendChild(_childItem);
-                                    });
+
+                        const dropdownButton = clone.querySelector('.dropdown button') || clone.querySelector('.dropdown-toggle');
+                        if (dropdownButton) {
+                            dropdownButton.addEventListener('click', function(e) {
+                                console.log('.dropdown button');
+                                e.preventDefault();
+                                e.stopPropagation();
+                                const parentElement = dropdownButton.closest('.dropdown');
+                                if (parentElement) {
+                                    parentElement.classList.toggle('open');
                                 }
-                                appendDivider();
-                            } else if (item.tagName === 'LI') {
-                                menuBottomUl.appendChild(item);
-                            }
-                        });
+                            });
+                        }
                     }
+                    // } else { // NOTE: This condition is never reached.  ???
```

Comment:
hi @dsuren1
I saw your code added here.
I can't replicate the use of the condition in else.
I have a menu configured by admin to have custom menus with sub-items.
But this block never seems to be used.
I'd like to make sure I'm not missing a specific condition in the menu configuration.